Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -240,302 +209,172 @@ function Set-FrontDoorRequiredTLSVersion
 
+    $FrontDoorDomainResources = @()
+
     # To keep track of remediated and skipped resources
     $logRemediatedResources = @()
     $logSkippedResources=@()
 
-    $controlIds = "Azure_FrontDoor_CDNProfile_DP_Use_Secure_TLS_Version_Trial"
-    
-    # No file path provided as input to the script. Fetch all Front Doors in the Subscription.
-    if($AutoRemediation)
+    $controlIds = "Azure_FrontDoor_CDNProfile_DP_Use_Secure_TLS_Version"
+
+    # Fetch all front doors in the Subscription.
+    if ([String]::IsNullOrWhiteSpace($FilePath))
     {
-        if(-not (Test-Path -Path $Path))
-        {
-            Write-Host "File containing failing controls details [$($Path)] not found. Skipping remediation..." -ForegroundColor $([Constants]::MessageType.Error)
-            Write-Host $([Constants]::SingleDashLine)
-            return
-        }
-        
-        Write-Host "Fetching all Front Doors failing for the [$($controlIds)] control from [$($Path)]..." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host "Fetching all Front Door CDN Profile Domain(s) in Subscription: [$($context.Subscription.SubscriptionId)]..." -ForegroundColor $([Constants]::MessageType.Info)
         Write-Host $([Constants]::SingleDashLine)
-        
-        $controlForRemediation = Get-content -path $Path | ConvertFrom-Json
-        $controls = $controlForRemediation.ControlRemediationList
-        $resourceDetails = $controls | Where-Object { $controlIds -eq $_.ControlId };
-        $validResources = $resourceDetails.FailedResourceList | Where-Object {![String]::IsNullOrWhiteSpace($_.ResourceName)}
-        
-        if(($resourceDetails | Measure-Object).Count -eq 0 -or ($validResources | Measure-Object).Count -eq 0)
+        # Get all Front Doors in a Subscription
+        $FrontDoorResources = Get-AzFrontDoorCdnProfile -SubscriptionId $SubscriptionId -ErrorAction Stop
+        foreach($frontdoor in $FrontDoorResources)
         {
-            Write-Host "No  Front Door(s) found in input json file for remediation." -ForegroundColor $([Constants]::MessageType.Error)
-            Write-Host $([Constants]::SingleDashLine)
-            return
-        }
-        $validResources | ForEach-Object {
-            $resource = $_
             try
             {
-                $resFrontDoor = Get-AzFrontDoorCdnProfile -SubscriptionId $SubscriptionId -Name $resource.ResourceName -ResourceGroupName $resource.ResourceGroupName  -ErrorAction SilentlyContinue
-                $FrontDoorResources += $resFrontDoor
+                $FrontDoorCustomDomains = @()
+
+                $FrontDoorCustomDomains = Get-AzFrontDoorCdnCustomDomain -ProfileName $frontdoor.Name -ResourceGroupName $frontdoor.ResourceGroupName 
+                
+                if($FrontDoorCustomDomains)
+                {
+                    Write-Host "Front Door CDN Profiles Configurations successfully fetched." -ForegroundColor $([Constants]::MessageType.Update)
```

Comment:
Let's also print profile name, else same line 10 times will be there on console.